ai-catalog/samples/gemini-live-todo/src/main/java/com/android/ai/samples/geminilivetodo/ui/TodoScreen.kt

```diff
@@ -0,0 +1,285 @@
+            MicButton(
+                uiState = uiState,
+                onToggle = { viewModel.toggleLiveSession(activity) },
+            )
+        },
+        floatingActionButtonPosition = FabPosition.Center,
+        modifier = Modifier.fillMaxSize(),
+    ) { paddingValues ->
+        Column(
+            modifier = Modifier
+                .padding(paddingValues)
+                .padding(16.dp)
+                .imePadding()
+                .fillMaxSize(),
+        ) {
+            TodoInput(
+                text = text,
+                onTextChange = { text = it },
+                onAddClick = {
+                    viewModel.addTodo(text)
+                    text = ""
+                },
+            )
+
+            when (uiState) {
+                is TodoScreenUiState.Initial -> {
+                    Box(
+                        modifier = Modifier
+                            .fillMaxSize(),
+                        contentAlignment = Alignment.Center,
+                    ) {
+                        CircularProgressIndicator()
+                    }
+                }
+                is TodoScreenUiState.Success -> {
+                    val todos = (uiState as TodoScreenUiState.Success).todos
+                    LazyColumn(modifier = Modifier.fillMaxSize()) {
+                        itemsIndexed(todos.reversed(), key = { index: Int, item: Todo -> item.id }) { index, todo ->
+                            TodoItem(
+                                modifier = Modifier,
+                                task = todo,
+                                onToggle = { viewModel.toggleTodoStatus(todo.id) },
+                                onDelete = { viewModel.removeTodo(todo.id) },
+                            )
+                            if (index != todos.size - 1) {
+                                HorizontalDivider()
+                            }
+                        }
+                    }
+                }
+                is TodoScreenUiState.Error -> {
+                    val todos = (uiState as TodoScreenUiState.Error).todos
+                    LazyColumn(modifier = Modifier.fillMaxSize()) {
+                        itemsIndexed(todos.reversed(), key = { index: Int, item: Todo -> item.id }) { index, todo ->
+                            TodoItem(
+                                modifier = Modifier,
+                                task = todo,
+                                onToggle = { viewModel.toggleTodoStatus(todo.id) },
+                                onDelete = { viewModel.removeTodo(todo.id) },
+                            )
+                            if (index != todos.size - 1) {
+                                HorizontalDivider()
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun TodoInput(text: String, onTextChange: (String) -> Unit, onAddClick: () -> Unit) {
```

Comment:
Every composable function except top level screen composables should take a modifier parameter